pkg/cri/server/container_create_linux.go

```diff
@@ -299,7 +299,8 @@ func (c *criService) containerSpecOpts(config *runtime.ContainerConfig, imageCon
 	specOpts = append(specOpts, customopts.WithAdditionalGIDs(userstr))
 
 	apparmorSpecOpts, err := generateApparmorSpecOpts(
```

Comment:
Since the Profile string is deprecated for both, instead than having the `generate*` handle this deprecation, it might be cleaner to just generate the `SecurityProfile` from the string if the profile is not set and then call the generate functions. Right now the logic to handle the different profiles is duplicated in those functions. I think it is cleaner to explicitly convert the deprecated type then only have one path to handle it.

Reply:
refactored the code.. will have to rebase